python/pyiceberg/io/pyarrow.py

```diff
@@ -28,12 +28,16 @@
 import os
 from abc import ABC, abstractmethod
 from functools import lru_cache, singledispatch
+from heapq import merge
```

Comment:
I think the changes in this file at least are nearly ready to go. We can probably get them in faster if we split out the changes in another PR.

Reply:
I think the logic for applying positional deletes in Arrow is about ready. I'd move that to a separate PR so we can get it in before we have the ability to track sequence numbers and do the planning operations.